maps_dashboards/public/components/layer_config/custom_map_config/custom_map_source.tsx

```diff
@@ -0,0 +1,308 @@
+  const [protocol, setProtocol] = useState(customMapProtocolOptions[1].value);
+  const [WMSLayers, setWMSLayers] = useState<string>('');
+  const [WMSVersion, setWMSVersion] = useState<string>('');
+  const [WMSFormat, setWMSFormat] = useState<string>('');
+  const [WMSStyles, setWMSStyles] = useState<string>('');
+  // CRS: Coordinate reference systems in WMS
+  const [WMSCRS, setWMSCRS] = useState<string>('');
+  const [WMSBbox, setWMSBbox] = useState<string>('');
+
+  const onChangeCustomMapURL = (e: any) => {
+    setCustomMapURL(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        url: e.target.value,
+      },
+    });
+  };
+
+  const onChangeCustomMapAttribution = (e: any) => {
+    setCustomMapAttribution(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        attribution: e.target.value,
+      },
+    });
+  };
+
+  const onChangeProtocol = (e: any) => {
+    setProtocol(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        protocol: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSLayers = (e: any) => {
+    setWMSLayers(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        layers: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSVersion = (e: any) => {
+    setWMSVersion(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        version: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSFormat = (e: any) => {
+    setWMSFormat(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        format: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSStyles = (e: any) => {
+    setWMSStyles(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        styles: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSCRS = (e: any) => {
+    setWMSCRS(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        crs: e.target.value,
+      },
+    });
+  };
+
+  const onChangeWMSBbox = (e: any) => {
+    setWMSBbox(e.target.value);
+    setSelectedLayerConfig({
+      ...selectedLayerConfig,
+      source: {
+        ...selectedLayerConfig?.source,
+        bbox: e.target.value,
+      },
+    });
+  };
+
+  const isInvalidURL = (url: string): boolean => {
+    if (url === '') return false;
+    try {
+      new URL(url);
+      return false;
+    } catch (e) {
+      return true;
```

Comment:
Shall we include error message from exception?

Reply:
this function It's only used to validate it's a valid URL not to validate it's map URL, we would just use it for EuiFormErrorText:
<img width="309" alt="image" src="https://user-images.githubusercontent.com/90288540/210405632-8433adcd-cda9-40cd-b7a6-4b47288bae05.png">

<img width="243" alt="image" src="https://user-images.githubusercontent.com/90288540/210405687-d94c60f1-b749-455b-b787-917a3a5267c8.png">